x-pack/auditbeat/processors/add_session_metadata/processdb/db.go

```diff
@@ -0,0 +1,742 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux
+
+package processdb
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"math/bits"
+	"os"
+	"path"
+	"slices"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/add_session_metadata/procfs"
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/add_session_metadata/timeutils"
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/add_session_metadata/types"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type TtyType int
+
+const (
+	TtyUnknown TtyType = iota
+	Pts
+	Tty
+	TtyConsole
+)
+
+type EntryType string
+
+const (
+	Init         EntryType = "init"
+	Sshd         EntryType = "sshd"
+	Ssm          EntryType = "ssm"
+	Container    EntryType = "container"
+	Terminal     EntryType = "terminal"
+	EntryConsole EntryType = "console"
+	EntryUnknown EntryType = "unknown"
+)
+
+var containerRuntimes = [...]string{
+	"containerd-shim",
+	"runc",
+	"conmon",
+}
+
+// "filtered" executables are executables that relate to internal
+// implementation details of entry mechanisms. The set of circumstances under
+// which they can become an entry leader are reduced compared to other binaries
+// (see implementation and unit tests).
+var filteredExecutables = [...]string{
+	"runc",
+	"containerd-shim",
+	"calico-node",
+	"check-status",
+	"conmon",
+}
+
+const (
+	ptsMinMajor     = 136
+	ptsMaxMajor     = 143
+	ttyMajor        = 4
+	consoleMaxMinor = 63
+	ttyMaxMinor     = 255
+)
+
+type Process struct {
+	Pids     types.PidInfo
+	Creds    types.CredInfo
+	CTty     types.TtyDev
+	Argv     []string
+	Cwd      string
+	Env      map[string]string
+	Filename string
+}
+
+var (
+	// The contents of these two files are needed to calculate entity IDs.
+	// Fail fast on startup if we can't read them.
+	bootID     = mustReadBootID()
+	pidNsInode = mustReadPidNsInode()
+	capNames   = []string{
+		"CAP_CHOWN",              // 0
+		"CAP_DAC_OVERRIDE",       // 1
+		"CAP_DAC_READ_SEARCH",    // 2
```

Comment:
Note that indexing can be used as strict commentary here. e.g. for the first three...
```suggestion
		0: "CAP_CHOWN", 
		1: "CAP_DAC_OVERRIDE", 
		2: "CAP_DAC_READ_SEARCH",
```
This way the comments can never go stale silently.

Reply:
Thanks, I didn't know about this, I've changed to use it